bofire/data_models/strategies/samplers/universal_constraint.py

```diff
@@ -0,0 +1,47 @@
+from typing import Literal, Type
+
+from bofire.data_models.constraints.api import (
+    LinearEqualityConstraint,
+    LinearInequalityConstraint,
+    NChooseKConstraint,
+    NonlinearEqualityConstraint,
+    NonlinearInequalityConstraint,
+)
+from bofire.data_models.features.api import (
+    ContinuousInput,
+    ContinuousOutput,
+    Feature,
+)
+from bofire.data_models.strategies.samplers.sampler import SamplerStrategy
+
+
+class UniversalConstraintSampler(SamplerStrategy):
+    """Sampler that generates samples by optimization in IPOPT.
+
+    Attributes:
+        domain (Domain): Domain defining the constrained input space
+        sampling_fraction (float, optional): Fraction of sampled points to total points generated in
+            the sampling process. Defaults to 0.3.
+        ipopt_options (dict, optional): Dictionary containing options for the IPOPT solver. Defaults to {"maxiter":200, "disp"=0}.
+    """
+
+    type: Literal["UniversalConstraintSampler"] = "UniversalConstraintSampler"
+    sampling_fraction: float = 0.3
+    ipopt_options: dict = {"maxiter": 200, "disp": 0}
```

Comment:
We will or have already these ipopt options dictionary for the DoEStrategy, would it make sense to model this also as a pydantic class, in which all possible options together with its defaults are listed?

Reply:
Hm, good question... I think most of the options that are available (here)[https://coin-or.github.io/Ipopt/OPTIONS.html] are only interesting for expert users, so I do not know for how many people this is relevant and if it is worth it to do the effort. Most people will probably only use the `maxiter` and `disp` options.  
However, for using the HSL linear solvers you have to set some options depending on your operating system, which can be annoying. The HSL linear solvers enormously reduce the computation time for large problems. If you think many users are interested in generating large models AND are willing to take the extra effort to install nondefault linear solvers, it might be worth it to handle the linear solver selection automatically in some way... What do you think? :D

Reply:
Hmm, I think I would let it as it is, and if necessary model it in a better way, what do you think?

Reply:
Agree.